logos/logos-workernode/tests/test_vllm_process.py

```
@@ -965,3 +1133,162 @@ def test_build_env_sets_persistent_caches(monkeypatch):
+
+# ---------------------------------------------------------------------------
+# _build_cmd integration — reasoning-parser + chat-template-kwargs
+# ---------------------------------------------------------------------------
+
+
+def test_build_cmd_gemma4_gets_reasoning_parser_and_chat_template_kwargs(
+    monkeypatch,
+) -> None:
+    """Gemma-4 with empty vllm_config: inferred reasoning-parser + inferred kwargs."""
+    handle = VllmProcessHandle("lane-test", 19000, OllamaConfig())
+    monkeypatch.setattr(handle, "_resolve_vllm_binary", lambda _configured: "/tmp/vllm")
```

Ruff (0.15.12)

[error] Probable insecure usage of temporary file or directory: "/tmp/vllm"

(S108)